adr/content-management/2023-01-06-upload-thumbnail-api-concept.md

```diff
@@ -0,0 +1,72 @@
+# 2023-01-06 - Upload Thumbnails API
+
+## Context
+We’re building an App that should optimize Thumbnails via the App. Currently, it’s not possible to upload Thumbnails via the API. All Thumbnails are processed by Shopware itself. There’s no way to upload your optimized Thumbnails via the API.
+
+However: Since it’s an App, we need to be able to do so via the API.
+
+## Decision
+### Migration
+Add a new column `isUploaded` into the media_thumbnail table. This flag lets us know if the thumbnail is uploaded or generated.
```

Comment:
This column should be in the MediaDefinition to enable this behavior per media and not per thumbnail record and rename it to `generate_thumbnails`. 
The default should be configurable using `shopware.yaml`, so we can allow disabling globally thumbnails may be necessary for other features.

Reply:
In our earlier idea, we want to keep everything of Shopware like default. The shop owner can upload their thumbnails via API, and then they decide whether to use them or not. And in case [Christopher mentioned here](https://github.com/shopware/platform/pull/2914/files#r1063351351), everything would work like normal if we uninstall the app.
 
 BTW, we need this change asap, @shyim could you help to review this PR again?  then we can have the final decision soon, and I will get started to implement this. Thank you so much for your attention and participation.

Reply:
So, for your case, the thumbnail generation would be still active, you just override it using API afterward?

Reply:
Yes, correct. Then there will be two thumbnails with the same size in `media_thumbnail` table (instead of only one as current), and the only difference is `isUploaded  =  true or false`